test/quick_test/test_images.py

```diff
@@ -0,0 +1,222 @@
+        Name of the test subject.
+
+    Raises
+    ------
+    AssertionError
+        If the image headers do not match
+    """
+
+    # Load images
+    test_subject = subjects_dir / test_dir / test_subject
+    test_image = load_image(test_subject, "brain.mgz")
+
+    reference_subject = subjects_dir / reference_dir / test_subject
+    reference_image = load_image(reference_subject, "brain.mgz")
+
+    # Get the image headers
+    headers = [test_image.header, reference_image.header]
+
+    # Check the image headers
+    header_diff = nibabel.cmdline.diff.get_headers_diff(headers)
+    assert header_diff == OrderedDict(), f"Image headers do not match: {header_diff}"
+    logger.debug("Image headers are correct")
+
+
+@pytest.mark.parametrize("test_subject", load_test_subjects())
+def test_seg_data(subjects_dir: Path, test_dir: Path, reference_dir: Path, test_subject: Path):
+    """
+    Test the segmentation data by calculating and comparing dice scores.
+
+    Parameters
+    ----------
+    subjects_dir : Path
+        Path to the subjects directory.
+    test_dir : Path
+        Name of test directory.
+    reference_dir : Path
+        Name of reference directory.
+    test_subject : Path
+        Name of the test subject.
+
+    Raises
+    ------
+    AssertionError
+        If the dice score is not 0 for all classes
+    """
+
+    test_file = subjects_dir / test_dir / test_subject
+    test_image = load_image(test_file, "aseg.mgz")
+
+    reference_subject = subjects_dir / reference_dir / test_subject
+    reference_image = load_image(reference_subject, "aseg.mgz")
+
+    labels = np.unique([np.asarray(reference_image.dataobj), np.asarray(test_image.dataobj)])
+
+    # Get the image data
+    test_data = np.asarray(test_image.dataobj)
+    reference_data = np.asarray(reference_image.dataobj)
+
+    # Compute the dice score
+    dscore = compute_dice_score(test_data, reference_data, labels)
+
+    # Check the dice score
+    np.testing.assert_allclose(
+        dscore, 0, atol=1e-6, rtol=1e-6, err_msg=f"Dice scores are not within range for all classes"
+    )
+
+    # assert dscore == 1, "Dice scores are not 1 for all classes"
+
+    logger.debug("Dice scores are within range for all classes")
+
+
+@pytest.mark.parametrize("test_subject", load_test_subjects())
+def test_int_data(subjects_dir: Path, test_dir: Path, reference_dir: Path, test_subject: Path):
+    """
+    Test the intensity data by calculating and comparing the mean square error.
+
+    Parameters
+    ----------
+    subjects_dir : Path
+        Path to the subjects directory.
+    test_dir : Path
+        Name of test directory.
+    reference_dir : Path
+        Name of reference directory.
+    test_subject : Path
+        Name of the test subject.
+
+    Raises
+    ------
+    AssertionError
+        If the mean square error is not 0
+    """
+
+    test_file = subjects_dir / test_dir / test_subject
+    test_image = load_image(test_file, "brain.mgz")
```

Comment:
Probably okay for now, but probably we want to test more than just one file, so how do we add more files here?